pkg/pillar/agentlog/logobject.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package agentlog
+
+import (
+	"encoding/json"
+	log "github.com/sirupsen/logrus"
+)
+
+// LogObject : Holds all key values pairs to be logged later
+type LogObject struct {
+	fields map[string]interface{}
+}
+
+// NewLogObject :
+// objType -> log, action, relation etc
+// objName -> App instance name, name of file being downloaded etc
+// objUUID -> UUID of the object if present (App instance UUID) or UUID of parent object
```

Comment:
Would be nice to have this be of type uuid.UUID so that folks don't accidentally use 3 strings

Reply:
Will make it UUID type.